extensions/caffeine/runtime/src/main/java/io/quarkus/caffeine/runtime/graal/CacheConstructorsFeature.java

```diff
@@ -82,4 +93,16 @@ public static String[] typesNeedingConstructorsRegistered() {
         };
     }
 
+    public static String[] typesNeedingConstructorsRegisteredWhenRecordingStats() {
+        return new String[] {
+                "com.github.benmanes.caffeine.cache.SILSMS",
+                "com.github.benmanes.caffeine.cache.SSSA",
+                "com.github.benmanes.caffeine.cache.SSLSA",
+                "com.github.benmanes.caffeine.cache.SSLSMS",
+                "com.github.benmanes.caffeine.cache.SSSMS",
+                "com.github.benmanes.caffeine.cache.SSSMSA",
+                "com.github.benmanes.caffeine.cache.SSSMSW",
+                "com.github.benmanes.caffeine.cache.SSSW"
```

Comment:
@ben-manes Could you please confirm that this list contains all caches that should be registered for reflection when the cache metrics are enabled?

Reply:
FWIW, it doesn't contain all of them. But I think it contains all the ones corresponding to the ones we already have.

Reply:
How did you determine which ones we need?

Reply:
If possible, you might write a unit test to assert that the bundled configurations work in native image. I don't know the class mappings offhand as it is done by logic and a test would avoid regressions if the listing became stale. 

I use Guava's `Sets.cartesianProduct(features)` which has a 2^n growth rate, so each new feature might double the class listings. In reality that is pruned down by reusing classes if they match the required data shape, e.g. an entry with key/value/timestamp can share a single class by either expireAfterAccess, expireAfterWrite, or variable expiration configurations. Caffeine's unit tests likewise use this cartesian product trick for parameterized tests to avoid regressions as features can interact in subtle ways.